cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DataRecoveryCommand.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.cli.admin.dataRecovery;
+
+import io.pravega.cli.admin.AdminCommand;
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.segmentstore.server.host.StorageLoader;
+import io.pravega.segmentstore.server.store.ServiceBuilder;
+import io.pravega.segmentstore.storage.StorageFactory;
+
+import java.io.File;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.logging.FileHandler;
+import java.util.logging.Formatter;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.FINE;
+import static java.util.logging.Level.FINER;
+import static java.util.logging.Level.INFO;
+import static java.util.logging.Level.SEVERE;
+
+/**
+ * Base for any data recovery related commands.
+ */
+public abstract class DataRecoveryCommand extends AdminCommand {
+    protected final static String COMPONENT = "storage";
+    protected Logger logger;
+
+    /**
+     * Creates a new instance of the DataRecoveryCommand class.
+     *
+     * @param args The arguments for the command.
+     */
+    DataRecoveryCommand(CommandArgs args) {
+        super(args);
+    }
+
+    /**
+     * Creates the {@link StorageFactory} instance by reading the config values.
+     *
+     * @param executorService   A thread pool for execution.
+     * @return                  A newly created {@link StorageFactory} instance.
+     */
+    StorageFactory createStorageFactory(ScheduledExecutorService executorService) {
+        ServiceBuilder.ConfigSetupHelper configSetupHelper = new ServiceBuilder.ConfigSetupHelper(getServiceBuilderConfig());
+        StorageLoader loader = new StorageLoader();
+        return loader.load(configSetupHelper, getServiceConfig().getStorageImplementation().toString(),
+                getServiceConfig().getStorageLayout(), executorService);
+    }
+
+    /**
+     * Creates logging directory and file for the command to be run. The path to the directory can be supplied on the
+     * command run. By default, the path is set as current user path.
+     *
+     * @param commandName   The name of the command to be run.
+     * @return              The path to the directory created.
+     * @throws Exception    In case of a failure in creating the directory or the file.
+     */
+    String setLogging(String commandName) throws Exception {
```

Comment:
This seems to be something not specific to this command, but something that is orthogonal to all commands, right? If you agree, maybe you could implement this in `AdminCommand`, so all other commands can benefit from logging if we configure to do so. What do you think?